### PR TITLE
Seed data, Identity modelling, new Markdown domain for RAN, validation for name

### PR DESCRIPTION
**Pull request by dave-cheong**

### CODE REVIEW

File: backend/typescript/models/name.model.ts [hunk 1]
```diff
@@ -8,15 +8,27 @@ import {IRAMObject, RAMSchema} from './base';
 const NameSchema = RAMSchema({
     givenName: {
         type: String,
-        trim: true
+        trim: true,
+        validate: {
```
Comment (sundriver) [inline backend/typescript/models/name.model.ts]: shouldn't the same rule be used for family name ?

File: backend/typescript/tests/name.model.spec.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    connectDisconnectMongo();
+    dropMongo();
+
+    beforeEach(async (done) => {
+        done();
+    });
+
+    it('inserts with given name', async (done) => {
+        try {
+            await NameModel.create({
```
Comment (sundriver) [inline backend/typescript/tests/name.model.spec.ts]: now that we have given name validation in place shouldn't this test fail

File: backend/typescript/models/name.model.ts (hunk 1 above)
Comment (dave-cheong) [inline backend/typescript/models/name.model.ts]: Ok moved to `pre validate` and throwing `Error` with no binding on a field.

File: backend/typescript/tests/name.model.spec.ts (hunk 2 above)
Comment (dave-cheong) [inline backend/typescript/tests/name.model.spec.ts]: Family name is optional per requirements and Brian's assertion, to support "mononyms" (eg Prince, Sting).

**Pull request merged**